content/blog/2025/anchor-position-area.md

```diff
@@ -0,0 +1,209 @@
+but we're choosing the container that we want to position our element inside of.
+
+## Positioning within the containing block
+
+Using `position-area` also handles some defaults so that the positioned element
+is placed near to the anchor, within its new containing block.
+
+First, any inset properties like `top` or `inline-start` that don't have values
+set on the positioned element resolve to `0`. This makes the containing block
+fill the entirety of its grid area.
+
+Then, `align-self` and `justify-self` are used to position the element close to
+the anchor, by specifying the alignment away from the specified grid area and
+towards the non-specified grid area. If the grid area is `center` or `span-all`,
+then the value is `anchor-center`.
+
+Together, that means that `position-area: top right` is equivalent to the
+following declaration using `anchor()`.
+
+```css
+.positioned-element {
+  bottom: anchor(top);
+  left: anchor(right);
+  top: 0;
+  right: 0;
+  justify-self: start;
+  align-self: end;
+}
+```
+
+{{ embed.codepen(
+  id='yyBmPLe',
+  title='position-area positioning',
+  user='jamessw',
+  height=600
+) }}
+
+## Shifts to prevent overflow
+
+As a side effect of being an absolutely positioned element with `justify-self`
+or `align-self` set, your positioned element will have the behavior of smartly
+shifting to stay within its original content block.
+
+If, after applying padding and margins, your positioned element would overflow
+the newly created containing block but would fit inside its original container
+block, it will get shifted to prevent overflow.
+
+In this example, the targets are all positioned with `position-area: center`.
+The top and bottom targets are shifted towards the center because they would
+otherwise overflow their original containing block, which is the `div` with the
+dashed border. They are shifted just enough to be flush with their containing
+block.
+
+{{ embed.codepen(
+  id='XJrLzWa',
+  title='position-area self alignment - absolute positioning',
+  user='jamessw',
+  height=350
+) }}
+
+Note that in that example, the targets are `position: absolute`. If instead, the
+targets are `position: fixed`, the targets do not overflow their original
+containing block (the viewport) and so no shifting happens.
+
+{{ embed.codepen(
+  id='ByBgbQa',
+  title='position-area self alignment - fixed position',
+  user='jamessw',
+  height=350
+) }}
+
+You can opt in to this behavior with `anchor()` by setting a value for all of
+`top`, `right`, `bottom`, and `left`, and setting a value for `justify-self` and
+`align-self`. But `postion-area` is much more terse.
+
+## Percentages
+
+One of the ways we observe a containing block is that percentages are resolved
+in relation to the containing block. That means a `padding: 5%` on an element
+positioned using `position-area` will have a padding of 5% of the size of the
+containing block, the section of the grid that was selected.
+
+{{ embed.codepen(
+  id='MYgMpMM',
```

Comment:
I think equal-size anchors would still help here, but not essential to see the difference.

Reply:
Oh right, thanks for the reminder!

Reply:
Great demo.